src/tests/users/signup.test.js

```
@@ -82,15 +82,13 @@ describe('signup', () => {
     return request({
       query: `
         mutation {
-          verifyEmail (verificationCode: "${users[0].verificationCode}") , record: {
-            emailVerified: true
-          })
+          verifyEmail (verificationCode: "${users[0].verificationCode}")
```

This doesn't test anything. It should accept at least 2 parameters:

1. Email address or username of the user
2. Input verification code to check against what's saved in the database

Also, no response data has been provided.

---

The `verifyEmail` resolver is not implemented as well

---

This test clearly fails and yet you're pushing your code. Are you disabling the `npm test` pre-push hook locally?

---

it was wrong mutation, I have updated mutation. It was working somehow. I dont know must have passed tests to push.

---

this is making user verified by setting emailVerified property to true. This is checking this is happening for any verification code.

---

So you're allowing an endpoint to set the `emailVerified` to be true? That's a big no-no. In that case, anyone can have their emails verified without going through the actual email verification process

---

if verification code is wrong this test will fail, is user is already verified then still this test will fail.

---

no need for email input since user will click link in email which will have verification code only I assume.

---

so on front end when verification email takes user to particular controller, then we can parse out verification code and make user verified if verification code exists.

---

The `verifyEmail` resolver is not defined so I don't have an idea that you're doing that. And this test should fail since that resolver is not defined.

---

this resolver is defined in verify-email.js in users module

---

https://github.com/kyleag/marketplays-api/blob/ayaz-users/src/modules/users/resolvers/verify-email.js

---

I just saw this one - wasn't there yet when I made the comment